src/block/test_generation.py

```diff
@@ -0,0 +1,34 @@
+# Python program to read
+# json file
+
+import json
+
+# Opening JSON file
+io = open('tests_bits_to_target_io.json')
+# Creating .cairo file
+f = open("test_bits_to_target.cairo", "w")
+
+# returns JSON object as 
+# a dictionary
+io_json = json.load(io)
+
+f.write("// This is a auto generated test\n")
+f.write("%lang starknet\n")
+f.write("from block_header import bits_to_target, target_to_bits\n")
+f.write("from starkware.cairo.common.cairo_builtins import BitwiseBuiltin\n")
+
+TAB = "    "
+tests_count = 1
+for io_case in io_json["bits_to_target"]:
```

Comment:
we could, eventually, iterate over different keys in the json, being the key the function to be called, and make different kind of tests depending on the key.